app/models/ability.rb

```diff
@@ -7,12 +7,18 @@ class Ability
   # See the wiki for details:
   # https://github.com/CanCanCommunity/cancancan/blob/develop/docs/define_check_abilities.md
   def initialize(user)
+    # Actions allowed for non-authenticated Users should add `skip_before_action :require_user`
+
+    # Start of Rules for all authenticated user with no additional roles required
     return if user.blank?
-    # Rules will go here.
 
-    # all authenticated
-    # can :view, :playground
+    can :manage, :detector__suggested_resource
+    can :manage, Detector::SuggestedResource
```

Comment:
Am I seeing this correctly? Both of these lines appear to be required, for different reasons? One grants access to the resource, while the other is needed to get access to the dashboard?

If it works, it works, but ... huh.

Reply:
It's complicated. I spent some time trying to understand how to avoid this, but it felt like it was going to require additional work in a possibly less maintainable way than just doubling up things in `Ability`. I wasn't sure how best to document it either.